src/sources/websocket/config.rs

```diff
@@ -0,0 +1,271 @@
+        value: String,
+    },
+}
+
+impl PongValidation {
+    pub fn matches(&self, msg: &str) -> bool {
+        match self {
+            PongValidation::Exact { value: expected } => msg == expected,
+            PongValidation::Contains { value: substring } => msg.contains(substring),
+        }
+    }
+}
+
+/// Configuration for the `websocket` source.
+#[serde_as]
+#[configurable_component(source("websocket", "Collect events from a websocket endpoint.",))]
+#[derive(Clone, Debug)]
+pub struct WebSocketConfig {
+    #[serde(flatten)]
+    pub common: WebSocketCommonConfig,
+
+    /// Decoder to use on each received message.
+    #[configurable(derived)]
+    #[serde(default = "default_decoding")]
+    pub decoding: DeserializerConfig,
+
+    /// Framing to use in the decoding.
+    #[configurable(derived)]
+    #[serde(default = "default_framing_message_based")]
+    pub framing: FramingConfig,
+
+    /// Number of seconds before timing out while connecting.
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[serde(default = "default_connect_timeout_secs")]
+    #[configurable(metadata(docs::advanced))]
+    #[configurable(metadata(docs::examples = 10))]
+    pub connect_timeout_secs: Duration,
+
+    /// Number of seconds before timing out while waiting for a reply to the initial message.
+    /// This is only used when `initial_message` is also configured.
+    #[serde_as(as = "serde_with::DurationSeconds<u64>")]
+    #[serde(default = "default_initial_message_timeout_secs")]
+    #[configurable(metadata(docs::advanced))]
+    #[configurable(metadata(docs::examples = 5))]
+    pub initial_message_timeout_secs: Duration,
+
+    /// An optional message to send to the server upon connection.
+    #[configurable(metadata(docs::advanced))]
+    #[configurable(metadata(docs::examples = "SUBSCRIBE logs"))]
+    #[serde(default)]
+    pub initial_message: Option<String>,
+
+    /// An optional application-level ping message to send over the WebSocket connection.
+    /// If not set, a standard WebSocket ping control frame is sent instead.
+    #[configurable(metadata(docs::advanced))]
+    #[serde(default)]
+    pub ping_message: Option<String>,
+
+    /// The expected application-level pong message to listen for as a response to a custom `ping_message`.
+    /// This is only used when `ping_message` is also configured. When a custom ping is sent,
+    /// receiving this specific message confirms that the connection is still alive.
+    #[configurable(metadata(docs::advanced))]
+    #[serde(default)]
+    pub pong_message: Option<PongMessage>,
+
+    /// The namespace to use for logs. This overrides the global setting.
+    #[configurable(metadata(docs::hidden))]
+    #[serde(default)]
+    pub log_namespace: Option<bool>,
+}
+
+const fn default_connect_timeout_secs() -> Duration {
+    Duration::from_secs(30)
+}
+
+const fn default_initial_message_timeout_secs() -> Duration {
+    Duration::from_secs(2)
```

Comment:
shouldn't this be closer to 30 too? if things are bad enough that connect takes 30s how can they be good enough that reply comes in 2?

Reply:
Yes, fixed it